packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2382,7 +2407,15 @@ export class KeyringController extends BaseController<
 
     try {
       const { type, data } = serialized;
-      return await this.#newKeyring(type, data);
+      const keyring = await this.#createKeyring(type, data);
+      this.#keyrings.push(keyring);
+      // If metadata is missing, assume the data is from an installation before
+      // we had keyring metadata.
+      if (this.#keyringsMetadata.length < this.#keyrings.length) {
+        console.log(`Adding missing metadata for '${type}' keyring`);
```

Comment:
Typically we avoid using `console.log` in libraries, instead preferring to pass a LogLevel logger into the constructor. This makes it easier for the client to control logging, to disable it if appropriate. You can see an example of this in the NetworkController: https://github.com/MetaMask/core/blob/ce1248513358d2651919d4e55d6e3fd9678a61f3/packages/network-controller/src/NetworkController.ts#L950

That seems like the nicer way to accomplish this.

But the idea here was that the console log could provide a hint if something unexpected happens. A standard troubleshooting step is to ask for console logs from users. They show up in Sentry breadcrumbs as well.